### PR TITLE
fix: notify assigned agent when customer replies via email

### LINKED ISSUE #360: Notify agents when a customer replies on an email-tagged ticket
## Problem

When a customer replies to a ticket by email, the poll appends the message as a comment on the existing work item — but no one on the team is notified. Agents only learn about the reply if they happen to refresh the ticket page or get to it through the queue.

For a support workflow this is a gap: the ticket is now \"awaiting agent action\" and we should actively pull the assigned agent's attention rather than wait for them to discover it.

## Expected

When the email-poll path adds a comment to an existing ticket (\`action: comment_added\` in the ingest result), the system should send a notification to the relevant internal users.

**Recipients (in priority order):**
1. The ticket's *Assigned To* user — primary recipient
2. If unassigned: the project's support team / on-call group (configurable)
3. CC anyone watching the ticket (future, if a watcher list is introduced)

**Channels — at minimum email; ideally configurable per-user:**
- Email (Microsoft Graph \`sendMail\` from the support mailbox — same path the customer already gets) — easiest first cut
- Microsoft Teams via Graph \`chatMessage\` / activity feed (future)
- In-app banner / unread badge on the ticket list (future)

**Email content:**
- Subject: \`[ZapDesk #N] New customer reply — \"<original subject>\"\`
- Body: customer's reply (plain text, signature-stripped — same render path as the comment)
- Link: \`View Ticket #N\` button to the ZapDesk UI
- Excluded from the threading prefix the customer sees — internal notifications are a separate thread

## Current Behaviour

\`/api/email/poll\` → \`ingestEmail\` → \`handleThreadReply\` → DevOps \`addComment\`. Done. No outbound notification of any kind.

## Acceptance Criteria

- [ ] Email-poll path triggers a notification when a comment is added to an existing email-tagged ticket
- [ ] The assigned-to user receives an email with the customer's reply and a link to the ticket
- [ ] Tickets without an assignee fall back to a configurable team/group address (e.g. \`SUPPORT_TEAM_NOTIFY_EMAIL\` env var, or a project description token)
- [ ] Notification is fire-and-forget — failure to notify must never block the comment add
- [ ] No notification is sent when the comment originates from an agent in the UI (only inbound customer replies)
- [ ] Internal-note style markers (\`[Internal Note] ...\`) are not relevant here since this path never produces them
- [ ] Setting can be opted out per-user (future) — for v1 a global setting is acceptable

## Out of Scope

- Teams / Slack / push notification channels — separate follow-ups
- Customer-side digest of \"someone responded to your ticket\" outside the standard agent-reply path — already covered

## Related

- #55 (parent — Email Channel Integration)
- #359 (sibling — full ticket history in customer-facing replies)

### PR DESCRIPTION
**Pull request by akash2017sky**
## Summary

When the email-poll path adds a comment to an existing ticket (`action: comment_added`), send an internal notification to the assigned agent so they don't have to discover the reply by refreshing the queue.

- Recipient resolution: `System.AssignedTo.uniqueName` if it looks like an email, otherwise the new `SUPPORT_TEAM_NOTIFY_EMAIL` env var, otherwise log and skip.
- Notification is **fire-and-forget** — wrapped so any mail send failure cannot block or fail the comment add.
- Only triggers from `handleThreadReply`. Agent-UI comments use a different code path and never produce a notification.
- Internal email goes on its own thread (no `In-Reply-To` header pointing at the customer's email), so it doesn't pollute the customer-facing conversation.
- Same `renderEmailBody` pipeline as the on-ticket comment, so signature stripping is consistent between the two views the agent sees.

## Fixes

Fixes #360

## What this covers

| Scenario | Before | After |
|---|---|---|
| Customer replies to a `[ZapDesk #N]`-tagged email; ticket has an assignee | Comment added silently — agent only sees it on next page refresh | Comment added; assignee receives `[ZapDesk #N] New customer reply — "<title>"` in their inbox with a `View Ticket #N` button |
| Customer replies; ticket is unassigned | Same silent comment | If `SUPPORT_TEAM_NOTIFY_EMAIL` is set, the team address gets the notification. Otherwise a `[Notify] … skipping` log line is emitted and the comment add still succeeds. |
| Assignee is a DevOps **team / group** identity (no `@` in `uniqueName`) | n/a (no notification existed) | Treated as no individual assignee — falls through to `SUPPORT_TEAM_NOTIFY_EMAIL`. |
| Mail send fails (PAT issue, Graph 5xx, network) | n/a | Logged with ticket id; comment add still succeeds, poll still returns `action: "comment_added"`. |
| Comment added by an agent in the ZapDesk UI | n/a | Unchanged — the UI uses a different DevOps code path; this notification never fires for agent-originated comments. |

## Files changed

- `src/lib/email-ingest.ts` — capture the work item returned by `addComment`, extract `System.AssignedTo` + `System.Title`, fire-and-forget `sendCustomerReplyNotification` call. Adds a `WorkItemFieldsResponse` type for the PATCH response shape and a `notifyAgentOfReply` helper.
- `src/lib/email.ts` — new `sendCustomerReplyNotification` exported. Uses `generateMessageId(id, 'agent-notify')` and deliberately omits `inReplyTo` so the agent thread is independent of the customer thread.
- `src/lib/email-templates.ts` — new `customerReplyNotificationTemplate` reusing the existing layout/branding.
- `CLAUDE.md` — documents the new optional `SUPPORT_TEAM_NOTIFY_EMAIL` env var.

## New env var

| Variable | Purpose | Required |
|---|---|---|
| `SUPPORT_TEAM_NOTIFY_EMAIL` | Fallback recipient when a ticket has no individual assignee. Set to a distribution list like `support@your-domain.com`. | No (without it, unassigned tickets just log a skip line) |

### Local manual (verified)

1. `bun run dev`, ensure `MAIL_*`, `AZURE_DEVOPS_PAT`, `EMAIL_WEBHOOK_SECRET`, `MAIL_POLL_MAILBOX` are set.
2. Create or pick a ticket the PAT can write to. Assign it to yourself in DevOps.
3. From a customer-domain address, send an email to the polled mailbox with `Subject: [ZapDesk #N] testing notification`.
4. Trigger the poll:
   ```powershell
   Invoke-RestMethod -Method Post -Uri http://localhost:3000/api/email/poll -Headers @{ "x-webhook-secret" = "<EMAIL_WEBHOOK_SECRET>" } | ConvertTo-Json -Depth 6
   ```
5. **Expected response:** `fetched: 1, ingested: 1`, `results[0].result.action: "comment_added"`, `ticketId: N`.
6. **Expected dev-server log:**
   ```
   Added email reply to ticket #N from <customer>
   [Email] Customer reply notification sent for ticket #N to <assignee>
   ```
7. **Expected assignee inbox:** new email subject `[ZapDesk #N] New customer reply — "<original ticket title>"`, body shows `Subject:`, `From:`, the customer reply (signature-stripped) and a green `View Ticket #N` button.
8. **Confirm thread isolation:** the notification email arrives in a *different mail thread* than the customer-facing emails for the same ticket.

### Fallback path (verified earlier in the same session)

9. Un-assign the ticket. Re-poll a fresh `[ZapDesk #N]` email.
10. **With `SUPPORT_TEAM_NOTIFY_EMAIL` set:** notification arrives at the team address.
11. **Without it set:** dev-server logs `[Notify] No agent assigned and SUPPORT_TEAM_NOTIFY_EMAIL not set — skipping notification for ticket #N`. Comment is still added — `action: "comment_added"` in the response.

### Failure path (graceful degradation)

12. Temporarily break `MAIL_CLIENT_SECRET`, send a `[ZapDesk #N]` email, re-poll.
13. **Expected:** poll response still `action: "comment_added"`; comment still on the ticket; dev-server logs `[Email] Failed to send customer reply notification for ticket #N: ...`. Restore secret afterwards.

### UI-originated comments

14. Add a comment from the ZapDesk ticket UI.
15. **Expected:** **no** `[Notify]` or `[Email] Customer reply notification` log line. The notification path is scoped strictly to inbound email replies.

### Automated checks

- [x] `bun run typecheck` — passes
- [x] `bun run lint` — passes (4 pre-existing warnings, unrelated)
- [x] `bunx prettier --check` on changed files — passes

## Out of scope

- Microsoft Teams / Slack / push notification channels (issue marks as future).
- In-app banner / unread badge.
- Per-user notification opt-out (issue accepts a global setting for v1).
- Watchers / CC list (issue notes "future, if a watcher list is introduced").
- Encrypted / S/MIME inbound mail.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/email-templates.ts [hunk 1]
```diff
@@ -131,6 +131,26 @@ export function agentReplyTemplate(opts: {
...
+  customerEmail: string;
+  replyContentHtml: string;
+}): string {
+  const ticketUrl = `${APP_URL}/tickets/${opts.ticketId}`;
+  return layoutWrapper(`
+    <p class="meta">New customer reply on ticket <strong>#${opts.ticketId}</strong></p>
+    <p class="meta"><strong>Subject:</strong> ${opts.ticketSubject}</p>
+    <p class="meta"><strong>From:</strong> ${opts.customerEmail}</p>
+    <div class="content">
+      ${opts.replyContentHtml}
```
[the remark this answers is not in the document]
Comment (akash2017sky) [reply src/lib/email-templates.ts:166]: Agreed, and fixed — but wider than the two fields, because the same hole sat two functions away.

Every template in `email-templates.ts` interpolated its plain-text fields raw, not just the notification this PR adds. `ticketConfirmationTemplate` and `statusChangeTemplate` do it with `subject` and `requesterName`, `statusBadge` with the status name, `agentReplyTemplate` with `agentName`, and `renderHistory` with `authorName`. All of those are reachable from the same inbound email anyone can send — and the confirmation and status-change ones go back **to the customer**, so escaping only the agent notification would have left the customer-facing path open.

`escapeText()` now covers all of them. The `*Html` fields are deliberately untouched, exactly as you say — `renderEmailBody` already produced those.

Covered by a new `src/lib/email-templates.test.ts`: an `` payload through the subject, requester name, customer address and status name, plus a case asserting `replyContentHtml` still comes through as live markup so the escaping did not overreach.

File: src/lib/email.ts [hunk 2]
```diff
@@ -277,6 +278,45 @@ export async function sendStatusChangeNotification(
...
+      ticketSubject,
+      customerEmail,
+      replyContentHtml,
+    });
+    await sendViaGraph({
+      to: agentEmail,
+      subject: `[ZapDesk #${ticketId}] New customer reply — "${ticketSubject}"`,
+      html,
+      messageId,
+    });
```
[the remark this answers is not in the document]
Comment (akash2017sky) [reply src/lib/email.ts:321]: Fixed — both calls now pass `signal: AbortSignal.timeout(GRAPH_TIMEOUT_MS)`, bounded at 15s.

Your point about `inFlight` is the sharper half: a hung token request does not just stall its own caller, it holds the shared promise so every later caller queues behind it indefinitely. And since both are awaited from fire-and-forget notification paths, nothing was ever going to notice.

On the runtime question — `package.json` pins `next@16.1.1`, which requires Node 20+, and `AbortSignal.timeout` has been available since Node 17.3. So it is safe here without a polyfill. Happy to add an explicit `engines.node` field if you want that pinned down rather than inferred from the Next version; it seemed worth raising separately rather than folding into this PR.

File: src/lib/email-templates.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+// anyone can send. Raw interpolation put that markup into mail delivered to an
+// agent, and back to the customer.
+const PAYLOAD = '<img src=x onerror="alert(1)">';
+
+describe('customerReplyNotificationTemplate', () => {
+  it('escapes the subject and the customer address', () => {
+    const html = customerReplyNotificationTemplate({
+      ticketId: 42,
+      ticketSubject: PAYLOAD,
+      customerEmail: `evil@example.com${PAYLOAD}`,
```
[the remark this answers is not in the document]
Comment (akash2017sky) [reply src/lib/email-templates.test.ts]: Done — both replaced with a plain `sender-address` placeholder. The tests only assert that the value gets escaped, so nothing needed to look like a real address.

Worth flagging that the guideline is breached elsewhere in the repo: `src/lib/sla.test.ts` has `client@example.com` and `test@example.com`, and `src/lib/effort-hours.test.ts` carries `test@example.com` on `main` — that last one I added during the #411 review, on advice to swap a real address for a placeholder, not realising a non-email placeholder was the house rule. Happy to sweep those in a small separate PR rather than drag unrelated files into this one.

File: src/lib/email-templates.ts
Comment (BenGWeeks) [inline src/lib/email-templates.ts:59]: Please use the ZapDesk iconography, not a new custom zap icon:

image

Comment (akash2017sky) [reply src/lib/email-templates.ts]: You're right, and the fix isn't here — it's in #378, which replaces this whole header with the real ZapDesk logo:

```html
<div class="header">
  <a href="${APP_URL}" style="text-decoration: none;">
    <img src="${ZAPDESK_LOGO_URL}" width="240" height="60" alt="${APP_NAME}" />
  </a>
</div>
```

I deliberately didn't duplicate that here — two branches editing the same header would just collide. I test-merged #378 into this branch to check the claim rather than assume it: `email-templates.ts` auto-merges and the `⚡` emoji is gone. So this resolves by inheriting #378 once that lands, with no code change needed in this PR.

Leaving the thread open until it's actually true on this branch — it isn't yet.

Worth flagging: that test-merge also showed conflicts in `email.ts`, `email-templates.test.ts` and `CLAUDE.md`. Not blocking, but #378 → main → here is the order that keeps them small, so #378 wants merging first.

**Comment by akash2017sky** [comment]
Rebased onto main and resolved the conflict in `email-ingest.ts`, where #376's attachment handling and this branch's agent notification had both rewritten `handleThreadReply`. Both sides are kept, with two judgement calls worth a reviewer's attention:

**Notify before linking attachments, not after.** The comment is already on the ticket by then, so a later attachment-link failure shouldn't cost the agent their alert.

**Inline images are stripped from the emailed copy.** This is the one behaviour neither PR had alone. After #376 the stored body points every inline image at a DevOps attachment URL, which needs a signed-in session — a mail client fetching one unauthenticated gets a 401 and draws a broken-image box. So the notification drops them and says how many were omitted; the agent follows the ticket link. The comment on the ticket is unaffected and still shows the images.

Added `email-ingest.reply.test.ts` covering the seam: it drives `ingestEmail` with `fetch` stubbed and asserts the URL is in the ticket comment and absent from the email. I verified it has teeth by reintroducing both failure modes — dropping the notification fails both tests, and passing the un-stripped body (the naive resolution) fails the second.

All checks green. The one open thread is your iconography comment, which I've replied to — the fix is #378, and I test-merged it to confirm this branch inherits it cleanly.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: src/lib/email-templates.ts
Comment (akash2017sky) [reply src/lib/email-templates.ts]: #378 has merged, and this branch now has it — done in 933eef3.

Checked rather than assumed. The custom zap is gone from `email-templates.ts` entirely:

```
$ grep -n "⚡" src/lib/email-templates.ts
(no matches)
```

and the header renders the real logo:

```html
<div class="header">
  <a href="${APP_URL}" style="text-decoration: none;">
    <img src="${ZAPDESK_LOGO_URL}" width="240" height="60" alt="${APP_NAME}" style="border: 0;" />
  </a>
</div>
```

`email-templates.ts` auto-merged with no conflict, as expected. Three files did conflict and are resolved: the `email.ts` import list needed both `customerReplyNotificationTemplate` and `layoutWrapper` since both are live; `email-templates.test.ts` was an add/add of two unrelated suites, now combined at 12 tests with none dropped; and the `CLAUDE.md` variable table took main version plus the one row this branch adds.

308 tests pass, typecheck clean, lint unchanged at the 5 pre-existing warnings.

**Comment by BenGWeeks** [comment]
@coderabbitai full review
